File: src/main/java/com/now/nowbot/throwable/TipsRuntimeException.java
package com.now.nowbot.throwable;

public class TipsRuntimeException extends RuntimeException implements BotException {
    byte[] image = null;
    public TipsRuntimeException(String msg){
        super(msg);
    }

    public TipsRuntimeException(byte[] image) {
        this.image = image;
    }

    @Override
    public boolean hasImage() {
        return image == null;
    }

    @Override
    public byte[] getImage() {
        return image;
    }
}

File: src/main/java/com/now/nowbot/throwable/TipsRuntimeException.kt
package com.now.nowbot.throwable

open class TipsRuntimeException : RuntimeException, BotException {
    final override var image: ByteArray? = null

    constructor(msg: String?) : super(msg)

    constructor(e: Throwable?) : super(e?.message)

    constructor(image: ByteArray?) {
        this.image = image
    }

    override fun hasImage(): Boolean {
        return image != null
    }
}
